### PR TITLE
Release/7.1.0

### PR DESCRIPTION
**Pull request by PhungNg**
# Changelog

## Added
- Patterns section
- Login Pattern
- Forms Pattern
- Usage with NPM - section to applicable components
- ```<nav>``` tags around Breadcrumbs ( #101 )
- Node package upgrade - Split code into various bundles which makes it possible to import specific components and color variables  ( #344 )

## Changed
- Landing page entry cards
- Toast component and documentation page
- Updated list of Products using DG

## Removed
- Disabled state on Links
- Styling rule on ```<small>``` tag ( #185 )
- Whitespace on Payex svg logo

## Fixed
- Alert boxes with no content, due to the last release, on Dialog page and datepicker page
- Togglebox focus state bug
- Issues with root-relative links when importing Stylesheet
- Text-default color. Is now #512b2b ( #386 )
- Weird looking top-border on Topbar( #340 )
- Bugs when nesting accordions. ( #429 )
- Scrolling issues with Dialog

### CODE REVIEW

**Comment by VikingTristan** [comment]
Hmm, seems like the main github action workflow is crashing due to some azure deployment bug(?).
Related https://github.com/Azure/login/issues/162

File: src/App/Patterns/content/Login/constants.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+export const guidelineList = [
+    <>The <b>Product label</b> should display the name of the product the user is about to log in to.</>,
+    <><b>Log in</b> should always be the title of the login form.</>,
+    <>A section of links to <b>third party logins</b> can be added. Use a divider to separate it from the standard login section. It can be placed either before or after depending on the most common or desired way to login to the specific portal. </>,
+    <>The <b>required input fields</b> varies depending on the type of login. The amount of required fields  differs depending on the authentication method. </>,
+    <><b>Remember me</b> is an optional field. When it is enabled it saves the data and present prefilled fields the next time the user is about to log in. </>,
+    <><b>Button</b> is used to allow the user to perform the login request.</>,
+    <><b>Additional help text</b> is optional and can be added underneath the button in order to provide the user with additional help or guidelines.</>,
+    <>Login forms should always have the <b>Swedbank pay logotype</b>.</>
+];
```
Comment (VikingTristan) [inline src/App/Patterns/content/Login/constants.js:12]: a list of react templates to enable html elements? cool :)

File: src/App/Patterns/content/Login/index.js [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+import { guidelineList } from "./constants";
+
+const basename = process.env.basename;
+
+const Overview = () => (
+    <section>
+        <h2 id="overview">Overview</h2>
+
+        <img src={`${basename}img/swedbankpay/documentation/patterns/login-pattern-example.png`} className="d-none d-md-block w-100 border-left border-right border-bottom" alt="Image for login screen" />
```
Comment (VikingTristan) [inline src/App/Patterns/content/Login/index.js:17]: This path doesn't exist when project is built:
`${basename}img/swedbankpay/documentation/patterns/login-pattern-example.png`

The culprit here is the brand. We might have to copy these assets to a brand folder in webpack.

File: src/App/Patterns/content/Patterns/index.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
+import React from "react";
+import { DocContainer } from "@docutils";
+import { Link } from "react-router-dom";
+import patternList from "@src/App/routes/patterns";
```
Comment (VikingTristan) [inline src/App/Patterns/content/Patterns/index.js:4]: *nitpick* we're importing routes here, for consistency with other files that import routes:
patternRoute/patternRoutes?

**Review by VikingTristan — COMMENTED** [review]
Looks very nice!

Haven't gotten through all of the files yet, but here are some comments so far:

There are a couple of things that needs to be done before this can be released imo.
- Patterns route doesn't exist ( try going to https://design.stage.swedbankpay.com/release/7.1.0/patterns/forms )
  - Check `generate-routes-copy-array.js`
- Some images (e.g. login example images) needs new paths or webpack needs an update
- New changelog says we're now releasing DG light, but I am not sure we are quite there yet. Change changelog?

**Comment by VikingTristan** [comment]
> Hmm, seems like the main github action workflow is crashing due to some azure deployment bug(?). Related [Azure/login#162](https://github.com/Azure/login/issues/162)

Fixed by using an older azcliversion

File: src/App/docutils/NpmInformation/index.test.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    it("is defined", () => {
+        expect(NpmInformation).toBeDefined();
+    });
+
+    it("renders", () => {
+        const wrapper = shallow(<NpmInformation/>);
+
+        expect(wrapper).toMatchSnapshot();
+        expect(wrapper.contains("undefined.init()")).toEqual(true);
+    });
```
Comment (VikingTristan) [inline src/App/docutils/NpmInformation/index.test.js:16]: Hmm :)

File: src/App/routes/getting-started.js [hunk 5]
```diff
@@ -6,14 +6,14 @@ module.exports = [
...
         path: "/get-started",
-        entryCardText: "Learn more about how to setup and use our system",
         redirect: "/get-started/introduction",
         appFolder: "GettingStarted",
         routes: [
             {
                 title: "Introduction",
                 path: "/get-started/introduction",
-                componentPath: "getting-started/Introduction"
+                componentPath: "getting-started/Introduction",
```
Comment (VikingTristan) [inline src/App/routes/getting-started.js:15]: *nitpick*
Consider renaming folder and component path to "get-started"

File: src/App/routes/patterns.js [hunk 6]
```diff
@@ -0,0 +1,35 @@
+module.exports = [
+    {
+        title: "Patterns",
+        icon: {
+            name: "dashboard"
+        },
+        path: "/patterns",
+        entryCardText: "Learn about how to we set up different structures",
+        redirect: "/patterns/patterns",
```
Comment (VikingTristan) [inline src/App/routes/patterns.js:9]: *nitpick for the future*
Should paths just be `/patterns` ?

**Review by VikingTristan — COMMENTED** [review]
Documentation says that toast should be top center on mobile. But, if I am not mistaken, it seems to be bottom left.
 
image

Also it refers to medium and larger screens as "web". Everything is "web" :)

File: src/less/components/toast.less [hunk 7]
```diff
@@ -2,106 +2,97 @@
...
                 color: @brand-info;
             }
         }
 
         &.toast-warning {
-            border-left: 3px solid @brand-warning;
-
             .material-icons {
-                color: @brand-warning;
+                color: #f35b1c;
```
Comment (VikingTristan) [inline src/less/components/toast.less:58]: Consider using a variable even though it's a classic "warning" color :)

File: src/less/components/toast.less [hunk 8]
```diff
@@ -2,106 +2,97 @@
...
                 margin-left: (-@base-margin-sm);
             }
 
             &:last-child {
                 cursor: pointer;
-                margin-left: @base-margin-sm;
+                margin-left: auto;
                 margin-right: (-@base-margin-sm);
-                color: lighten(@text-color, 20%);
+                color: @medium-brown;
```
Comment (VikingTristan) [inline src/less/components/toast.less:88]: Brown color might not be suited for the payex brand, but I guess it doesn't matter too much atm.

File: src/scripts/main/dialog/index.js [hunk 9]
```diff
@@ -82,6 +82,12 @@ class Dialog {
                 this.close();
             });
         }
+
+        window.addEventListener("popstate", () => {
+            if (document.body.className.includes("dialog-open")) {
+                document.body.classList.remove("dialog-open");
+            }
+        });
```
Comment (VikingTristan) [inline src/scripts/main/dialog/index.js:90]: nice

File: src/scripts/main/toast/index.js [hunk 10]
```diff
@@ -118,6 +122,8 @@ class Toast {
         // Set content
         toastContent.classList.add("toast-content");
         toastContent.innerHTML = this.message;
+        toastContent.setAttribute("role", "alert");
```
Comment (VikingTristan) [inline src/scripts/main/toast/index.js:125]: whoah very nice

File: src/App/ComponentsDocumentation/components/Pagination/index.js [hunk 11]
```diff
@@ -3,6 +3,7 @@ import React from "react";
 import { ComponentPreview, DocContainer } from "@docutils";
 import PaginationComponent from "@components/Pagination";
 import CodeTags from "@components/CodeTags";
+import NpmInformation from "~/src/App/docutils/NpmInformation";
```
[the remark this answers is not in the document]
Comment (eskilsen) [reply src/App/ComponentsDocumentation/components/Pagination/index.js]: 🚀  Done

File: src/App/ComponentsDocumentation/components/Sidebar/index.js [hunk 12]
```diff
@@ -7,6 +7,7 @@ import CodeTags from "@components/CodeTags";
 import { sidebar } from "@src/scripts/main";
 
 import { SidebarNavList, SidebarNavListExtended } from "./constants";
+import NpmInformation from "~/src/App/docutils/NpmInformation";
```
[the remark this answers is not in the document]
Comment (eskilsen) [reply src/App/ComponentsDocumentation/components/Sidebar/index.js]: Donolini 🚀

File: src/App/Home/constants.js [hunk 13]
```diff
@@ -1,4 +1,9 @@
 export const changeLogs = [
+    {
+        version: "7.1.0",
+        title: "Design Guide Light and new Pattern section",
+        text: "Introducing Design Guide Light, a NPM upgrade which makes it possible to import specific component styling and script needed for your project - enjoy the flexibility y’all! Documentation wise, we added a new Pattern section to showcase best practice solutions for common product features, starting off with Login and Forms page. We also solved a bunch of forgotten GitHub issues and updated the Toast component."
```
[the remark this answers is not in the document]
Comment (VikingTristan) [reply src/App/Home/constants.js:5]: Perhaps something along the lines of:
"One step closer to a lighter and a more component based Design Guide"

[the remark this answers is not in the document]
Comment (eskilsen) [reply src/App/Home/constants.js:5]: !Importante con 7.1.0 senior!

**Review by VikingTristan — APPROVED** [review]
Very nice!

File: src/App/docutils/NpmInformation/index.test.js (hunk 4 above)
Comment (eskilsen) [reply src/App/docutils/NpmInformation/index.test.js:16]: Fixed 🚀

File: src/less/components/toast.less (hunk 7 above)
Comment (eskilsen) [reply src/less/components/toast.less:58]: 🚀  done

File: src/App/routes/getting-started.js (hunk 5 above)
Comment (eskilsen) [reply src/App/routes/getting-started.js:15]: 🚀  Donolini habibi

File: src/App/Patterns/content/Login/index.js (hunk 2 above)
Comment (VikingTristan) [reply src/App/Patterns/content/Login/index.js]: I believe we fixed this

File: src/App/Home/constants.js (hunk 13 above)
[the remark this answers is not in the document]
Comment (eskilsen) [reply src/App/Home/constants.js]: 🚀  Done

File: src/App/ComponentsDocumentation/components/Dialog/index.js [hunk 14]
```diff
@@ -22,11 +23,7 @@ const HowItWorks = () => (
             The dialog body should contain the main content, usually just text confirming an action.
             In the footer you will find buttons to confirm or cancel the requested action.
         </p>
-        <AlertComponent type="warning">
-            <h4>Initializing dialog buttons</h4>
-            To make a button open the dialog box you need to
-            include <CodeTags type="secondary" code={"data-dialog-open=\"{#your-dialog-id}\""} /> as an attribute to that button.
-        </AlertComponent>
+        <AlertComponent type="warning" icon="warning" text={<p><b>Initializing dialog buttons:</b> To make a button open the dialog box you need to include <CodeTags type="secondary" code={"data-dialog-open=\"{#your-dialog-id}\""} /> as an attribute to that button.</p>} />
```
[the remark this answers is not in the document]
Comment (eskilsen) [reply src/App/ComponentsDocumentation/components/Dialog/index.js:26]: Done 🚀

**Pull request merged**